core/frontend/src/tile/map/MapLayerImageryFormats.ts

```diff
@@ -73,51 +83,50 @@ export abstract class MapLayerImageryProvider {
     this._testChildAvailability(tile, resolveChildren);
   }
 
-  public async getToolTip(strings: string[], quadId: QuadId, _carto: Cartographic, tree: ImageryMapTileTree): Promise<void> {
-    if (debugToolTip) {
-      const cartoRectangle = tree.cartoRectangleFromQuadId(quadId);
-      const debugString = `QuadId: ${quadId.debugString} Rectangle: ${cartoRectangle.latLongString} EPSG:3857: ${this.getEPSG3857ExtentString(quadId.row, quadId.column, quadId.level)} URL: ${this.constructUrl(quadId.row, quadId.column, quadId.level)}`;
-      strings.push(debugString);
-      // eslint-disable-next-line no-console
-      console.log(debugString);
-    }
+  public async getToolTip(_strings: string[], _quadId: QuadId, _carto: Cartographic, _tree: ImageryMapTileTree): Promise<void> {
   }
-  private getRequestAuthorization(): RequestBasicCredentials | undefined {
+
+  protected getRequestAuthorization(): RequestBasicCredentials | undefined {
     return (this._settings.userName && this._settings.password) ? { user: this._settings.userName, password: this._settings.password } : undefined;
   }
 
-  // returns a Uint8Array with the contents of the tile.
-  public async loadTile(row: number, column: number, zoomLevel: number): Promise<ImageSource | undefined> {
-    const tileUrl: string = this.constructUrl(row, column, zoomLevel);
-    if (tileUrl.length === 0)
+  protected getImageFromTileResponse(tileResponse: Response, zoomLevel: number) {
+    const byteArray: Uint8Array = new Uint8Array(tileResponse.body);
+    if (!byteArray || (byteArray.length === 0))
+      return undefined;
+    if (this.matchesMissingTile(byteArray) && zoomLevel > 8)
       return undefined;
+    let imageFormat: ImageSourceFormat;
+    switch (tileResponse.header["content-type"]) {
+      case "image/jpeg":
+        imageFormat = ImageSourceFormat.Jpeg;
+        break;
+      case "image/png":
+        imageFormat = ImageSourceFormat.Png;
+        break;
+      default:
+        return undefined;
+    }
 
+    return new ImageSource(byteArray, imageFormat);
+  }
+
+  // returns a Uint8Array with the contents of the tile.

```

Comment:
No it doesn't, it returns a ImageSource | undefined.

Reply:
Comment was there prior to my changes, I can remove it if needed.

Reply:
comment removed.